go/merkletree/merkle_verifier.go

```diff
@@ -0,0 +1,176 @@
+package merkletree
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+)
+
+// MerkleVerifier is a class which knows how to verify merkle inclusion and consistency proofs.
+type MerkleVerifier struct {
+	treeHasher *TreeHasher
+}
+
+// NewMerkleVerifier returns a new MerkleVerifier for a tree based on the passed in hasher.
+func NewMerkleVerifier(h HasherFunc) MerkleVerifier {
+	return MerkleVerifier{
+		treeHasher: NewTreeHasher(h),
+	}
+}
+
+// VerifyInclusionProof verifies the correctness of the passed in proof given the passed in information about the tree and leaf.
+func (m MerkleVerifier) VerifyInclusionProof(leafIndex, treeSize int64, proof [][]byte, root []byte, data []byte) error {
```

Comment:
It isn't clear what `data` is. Perhaps rename it to `leaf` or something like that?